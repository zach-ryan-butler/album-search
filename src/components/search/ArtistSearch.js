import React from 'react';
import PropType from 'prop-types';

function ArtistSearch({ text, textHandler, submitHandler }) {
  return (
    <form onSubmit={submitHandler}>
      <input
        type="text" 
        value={text} 
        placeholder="search.." 
        onChange={textHandler} /> 
    </form>
  );
} 

ArtistSearch.propTypes = {
  text: PropType.string.isRequired,
  textHandler: PropType.func.isRequired,
  submitHandler: PropType.func.isRequired
};

export default ArtistSearch;
