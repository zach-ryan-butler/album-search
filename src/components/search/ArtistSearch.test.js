import React from 'react';
import { shallow } from 'enzyme';
import ArtistSearch from './ArtistSearch';

describe('ArtistSearch component', () => {
  it('renders ArtistSearch', () => {
    const wrapper = shallow(<ArtistSearch
      text='some text'
      textHandler={() => {}}
      submitHandler={() => {}}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
