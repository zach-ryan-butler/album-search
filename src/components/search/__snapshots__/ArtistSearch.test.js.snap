// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistSearch component renders ArtistSearch 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="search.."
    type="text"
    value="some text"
  />
  <button
    disabled={false}
  >
    Search
  </button>
</form>
`;
